packages/lodestar/src/api/impl/validator/index.ts

```diff
@@ -133,6 +135,15 @@ export function getValidatorApi({chain, config, logger, metrics, network, sync}:
     }
   }
 
+  function notOnOptimisticBlockRoot(beaconBlockRoot: Root): void {
```

Comment:
Currently this is applied to:
- `produceBlock`
- `produceSyncCommitteeContribution`

Should it be applied to attestations too? Could you add a nice long comment at the top of `notOnOptimisticBlockRoot` explaining why we don't allow this actions?

Reply:
you mean we are doing it on `produceAttestationData` and `produceSyncCommitteeContribution` only and not on `produceBlock`? 
`produceBlock` when calls assembleBody, which in turn call EL's fetcg payload sequence (notify/getpayload), the EL will revert with Syncing there. The reason it has not been done here is for a optimization that EL could get synced between the last time it responded syncing and this opportunity of produceBlock.

Reply:
My bad, miss-understood the code. Current approach looks good!